### PR TITLE
ancillary fix dragonfly build

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Review by dtolnay — APPROVED** [review]

**Comment by rtzoeller** [comment]
@devnexen it doesn't look like this fixes all of the compilation errors introduced by #91553. `library/std/src/sys/unix/net.rs` was modified and `libc::SO_PASSCRED` is now used, but rust-lang/libc doesn't provide a definition of `SO_PASSCRED` for DragonFly (although DragonFly does define it).

**Comment by devnexen** [comment]
@dtolnay can we use the last commit instead please ? :-) thanks in advance.

File: library/std/src/os/unix/net/ancillary.rs [hunk 1]
```diff
@@ -249,46 +249,46 @@ impl SocketCred {
     #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
     #[must_use]
     pub fn new() -> SocketCred {
-        SocketCred(libc::cmsgcred { cmsgcred_pid: 0, cmsgcred_uid: 0, cmsgcred_gid: 0 })
+        SocketCred(libc::cmsgcred { cmcred_pid: 0, cmcred_uid: 0, cmcred_gid: 0 })
```
Comment (rtzoeller) [inline library/std/src/os/unix/net/ancillary.rs:252]: This still appears wrong to me. DragonFly defines additional members `cmsgcred` which aren't being referenced here, leading to [E0063](https://doc.rust-lang.org/error-index.html#E0063) when I build this locally. https://github.com/rust-lang/libc/blob/9b054d8779d2f71f1803e20f0e3b3327a4ae2764/src/unix/bsd/freebsdlike/mod.rs#L214-L221

@devnexen are you building these changes locally? My understanding is with DragonFly being a Tier 3 target, this code isn't actually built or tested in PR or CI builds, meaning you can't rely on bors to validate your changes.

I can walk you through building this locally if needed, but I'm concerned that neither this PR nor https://github.com/rust-lang/rust/commit/e68887e67cc6b7bb4ea5113a40eaa4c0831bda13 were ever actually tested...

Comment (devnexen) [reply library/std/src/os/unix/net/ancillary.rs:252]: is that the only build error ?

Comment (rtzoeller) [reply library/std/src/os/unix/net/ancillary.rs:252]: It is the only error which is not originating as a warning.

Are you set up to build this locally to test your changes?

Comment (devnexen) [reply library/std/src/os/unix/net/ancillary.rs]: is it building for you ? unfortunately I lost my DragonFly VM so I can t test for now.

Comment (devnexen) [reply library/std/src/os/unix/net/ancillary.rs]: sorry for the disturbance, for now better to revert at least that will unblock your situation.

**Pull request closed without merging**